martin/src/pg/table_source.rs

```diff
@@ -193,6 +203,38 @@ FROM {schema}.{table};
         .and_then(|p| polygon_to_bbox(&p)))
 }
 
+async fn calc_id_column(
```

Comment:
I don't think you should query DB for the available IDs. `TableInfo.properties` already contains all available columns and their types - you can simply look through that HashMap and see if any of the requested ID columns are present.